pylint/checkers/spelling.py

```diff
@@ -233,48 +260,39 @@ def open(self):
         self.initialized = False
         self.private_dict_file = None
 
-        if enchant is None:
+        if not hunspell:
             return
-        dict_name = self.config.spelling_dict
-        if not dict_name:
+        dict_tuple = _find_dictionary(
+            self.config.spelling_dict_name, self.config.spelling_dict_paths
+        )
+        if not dict_tuple:
             return
+        self.hobj = hunspell.HunSpell(dict_tuple[0], dict_tuple[1])
 
-        self.ignore_list = [
+        self.ignore_list = {
             w.strip() for w in self.config.spelling_ignore_words.split(",")
-        ]
+        }
         # "param" appears in docstring in param description and
         # "pylint" appears in comments in pylint pragmas.
-        self.ignore_list.extend(["param", "pylint"])
+        self.ignore_list.update(["param", "pylint"])
 
         # Expand tilde to allow e.g. spelling-private-dict-file = ~/.pylintdict
         if self.config.spelling_private_dict_file:
-            self.config.spelling_private_dict_file = os.path.expanduser(
-                self.config.spelling_private_dict_file
+            # TODO if/once hunspell starts supporting adding hunspell "personal" dicts
+            #      using hobj.add_dic, refactor this to support that API
+            self.private_dict_file = open(
+                os.path.expanduser(self.config.spelling_private_dict_file), "a+"
             )
-
-        if self.config.spelling_private_dict_file:
-            self.spelling_dict = enchant.DictWithPWL(
-                dict_name, self.config.spelling_private_dict_file
-            )
-            self.private_dict_file = open(self.config.spelling_private_dict_file, "a")
-        else:
-            self.spelling_dict = enchant.Dict(dict_name)
+            self.private_dict_file.seek(0)
+            for line in self.private_dict_file.read().splitlines():
+                self.ignore_list.add(line)
 
         if self.config.spelling_store_unknown_words:
             self.unknown_words = set()
 
         self.tokenizer = get_tokenizer(
-            dict_name,
-            chunkers=[ForwardSlashChunkder],
-            filters=[
-                EmailFilter,
-                URLFilter,
-                WikiWordFilter,
-                WordsWithDigigtsFilter,
-                WordsWithUnderscores,
-                CamelCasedWord,
-                SphinxDirectives,
-            ],
+            filters=[EnglishWordFilter(), WikiWordFilter(), CamelCasedWordsFilter()],
```

Comment:
Does hunspell automatically filters out the same tokens as the previous incantation, e.g. `URLFilter` and friends?